contrib/devtools/Dockerfile

```diff
@@ -2,7 +2,7 @@
 # docker build --pull --rm -f "contrib/devtools/Dockerfile" -t cosmossdk-proto:latest "contrib/devtools"
 # docker run --rm -v $(pwd):/workspace --workdir /workspace cosmossdk-proto sh ./scripts/protocgen.sh
 
-FROM bufbuild/buf:1.1.0 as BUILDER
+FROM bufbuild/buf:1.7.0 as BUILDER
```

Comment:
Have you checked changing this won't fail some scripts?
https://github.com/cosmos/cosmos-sdk/blob/main/scripts/protocgen-any.sh#L11 I believe this should be changed to `buf alpha protoc` for instance if we upgrade.

Reply:
I've only tested `make proto-all` which does `proto-format proto-lint proto-gen`

The `make proto-any` is still failing with:
```
roman@akhtariev ~/projects/cosmos-sdk (roman/fix-proto-gen) $ make proto-gen-any
Makefile:74: RocksDB support is disabled; to build and test with RocksDB support, set ENABLE_ROCKSDB=true
Generating Protobuf Any
third_party/proto/google/protobuf/any.proto:35:8:gogoproto/gogo.proto: does not exist
make: *** [Makefile:409: proto-gen-any] Error 100
```
when I do `buf alpha protoc`. Any ideas?

Reply:
IMHO It's fine to leave it like that, at least it is not breaking with a command not found due to this PR.

The whole third_party folder will be soon gone anyway.

Reply:
`make proto-swagger-gen` is working

Reply:
`make proto-check-breaking` is not working but I suspect that it is because the `main` is currently on the wrong version:
```
roman@akhtariev ~/projects/cosmos-sdk (roman/fix-proto-gen) $ make proto-check-breaking
Makefile:74: RocksDB support is disabled; to build and test with RocksDB support, set ENABLE_ROCKSDB=true
Failure: buf.build/googleapis/googleapis:40f07f5b563941f2b20b991a7aedd53d: does not exist
make: *** [Makefile:427: proto-check-breaking] Error 1
```

Here, we are comparing against `main`. `main` is on `40f07f5b563941f2b20b991a7aedd53d` hash that is failing

Reply:
So I think we should merge this and try `make proto-check-breaking` again

Reply:
I think these are the only relevant scripts. Is there anything else I should check? @julienrbrt

Reply:
Yes, I think we should update it here too: https://github.com/cosmos/cosmos-sdk/blob/1c8a2d90691aeff38a5baa51b6d3edc4acd9566e/client/v2/internal/buf.lock#L23 and it's good to go!

Reply:
done!